ledger/ledger.go

```diff
@@ -836,15 +836,19 @@ func (l *Ledger) VerifiedTransactionCache() verify.VerifiedTransactionCache {
 // If a value of zero or less is passed to maxTxnBytesPerBlock, the consensus MaxTxnBytesPerBlock would
 // be used instead.
 // The tracer argument is a logic.EvalTracer which will be attached to the evaluator and have its hooked invoked during
-// the eval process for each block. A nil tracer will skip tracer invocation entirely.
+// the eval process for each block. A nil tracer will default to the tracer attached to the ledger.
 func (l *Ledger) StartEvaluator(hdr bookkeeping.BlockHeader, paysetHint, maxTxnBytesPerBlock int, tracer logic.EvalTracer) (*eval.BlockEvaluator, error) {
+	tracerForEval := tracer
```

Comment:
Stylistic and ignoreable nit: introducing a new variable isn't necessary. You can either rename the param `tracerForEval` or even without renaming simply overwrite it with `l.tracer` when it's `nil`.

Reply:
Another point to consider. Currently, and in the pre-existing code the `tracer` param overrides `l.tracer` even when it's non-nil. This feels like an error to me and perhaps deserves an error (though it may very well break some tests and even live uses). Something like:
```go
if tracer != nil && l.tracer != nil {
    return nil, fmt.Errorf("ledger StartEvaluator(): ambiguous non-nil EvalTracers provided to both in Ledger and as tracer param")
}

Reply:
We need to make sure this doesn't break the simulate endpoint, which attaches its own tracer with this function.

Reply:
@jasonpaulos are there regression tests?

Reply:
I don't believe there are any tests of simulate with the config option `EnableTxnEvalTracer=true` yet. This would be a good test to ensure simulate isn't affected by this PR.

Probably the simplest way to achieve this would be to either augment or copy [`TestSimulateTransaction`](https://github.com/algorand/go-algorand/blob/32466b9a26fdef8288c7f2b1cad77a1bd5946a26/test/e2e-go/restAPI/restClient_test.go#L1571C6-L1571C29) to also test with the new `DevModeTxnTracerNetwork.json`.

Reply:
Yeah the intention right now is to allow `StartEvaluator` to have the final say in which tracer is attached to the newly created evaluator. The ledger's tracer is provided as the default in case nothing is passed in.

This lets simulate/txn eval tracer work for their respective use cases currently, but maybe we would want to revisit this decision in the future if/when we have more tracers competing for access to the evaluator.

Reply:
Added a test which is a copy of the TestSimulateTransaction test w/ a couple of assertions changed because devmode lets us be confident that the round increases deterministically.

Reply:
Very nice!